### ISSUE TITLE
Version 2.0.0

### ISSUE THREAD

**Issue by finnp** [finnp/png-itxt#4]
Hey @briancullen ,

wow, thanks for all your work for this module! The new features are pretty great!

I just had a short look into it and pushed a few changes. I changed the style to [standard](https://www.npmjs.com/package/standard) and added it to the project. That's my preferred way for code styling.  Also I removed the compatibility mode because I think it's not needed.  Also I removed the `dist` folder (we need to change the readme accordingly). The reason is that I generally expect that people who use this as a browser library can use `browserify` or other preferred tools for themselves. It shouldn't be part of this repository. Generally though I think it's super awesome that it works in the browser now! One of the reasons I developed this was to use it in the browser.

Also I think the old interface `pngitxt.set(key, value)` should be readded. I like the simplicity of it and it does not collide with `pngitxt(obj)`, because we could test if the `obj` is a string and if so use the old interface. Also while it's great that the ReadMe has so much detail, I think it should be reordered a bit so that easy to use examples should be in the beginning and more detailed explanations and usages in the end.

This is just my first thoughts about it. I'll look into it in more detail later. Thanks already for all your great work!

**Comment by briancullen** [comment]
No problems with removing the compatibility stuff. I didn't know you were explicitly using a particular standard so apologies if that caused you a lot of clean up - coming from a Java/C background not using semi colons feels really weird.

As for the dist folder I can see what you mean in terms of people should be able to do it themselves but I did think that an example of how it could be used was useful.

As for reinstating the old set interface you are right that would be relatively easy to do but I won't get a chance to do it properly until next week at this stage as I'm away for Christmas.

Anyway let me know what you think of the rest of it and then when everything else is settled I'll go over the readme.

**Comment by briancullen** [comment]
I have just pushed some changes to the set interface to a new branch called version2 - just to keep it separate from the main branch until you are happy with the changes.

You can now use `pngitxt.set("cat", "cute")` rather than having to use an object. The chunk type defaults to iTXt as it isn't specified.

The other change I made was to change the boolean replaceAll flag with an object. At the moment all the object is checked for is the replaceAll flag but I wanted to leave it open for additions in the future. In particular I keep thinking that you may not always want to replace all the existing chunks of the same type - as example of this could be where a copyright notice is given in different languages.

**Comment by briancullen** [comment]
Just wanted to bump this to see if you still want to make a version 2.

Anyway I was playing around a while ago with the different options and I changed the object for the set operation back to a number which indicated the type of replace to do. There are constants defined for each type:
```
pngtxt.REPLACETYPE  // replace chunks of same type and keyword.
pngtxt.REPLACEONE // replaces the first chunk of the same type and keyword.
pngtxt.REPLACELANG // replaces iTXt blocks with the same keyword and language
pngtxt.REPLACEALL // replaces any text chunks with same keyword
pngtxt.REPLACENONE // basically append - not sure it shouldn't be renamed.
```
So for example you could  write `pngitxt.set('cat', 'cute', pngitxt.REPLACEALL)` to replace all the text chunks with the keyword cat. The only issue with this is in implementing it the most straightforward way means that the program always puts the updated chunk right before the end chunk - not sure if the position in the file makes a difference. The specification says it doesn't matter but I don't know.

**Comment by finnp** [comment]
Hey @briancullen! Sorry I am a bit busy lately. If you want to go ahead and publish 2.0.0 yourself I am cool with that. You would just need to tell me your npm handle and I can add you as a maintainer there.  Otherwise please bump this again in a couple of days :grin:

**Comment by finnp** [comment]
I just released 2.0.0 to not confuse anybody
